src/Microsoft.Azure.SignalR.Common/Auth/JwtBuilders.cs

```diff
@@ -0,0 +1,436 @@
+                    Base64.GetMaxEncodedToUtf8Length(headerSha256.Length + payloadLength)
+                    + 1 // dot
+                    + Base64.GetMaxEncodedToUtf8Length(32); // signature SHA256 hash size
+            }
+            _memoryStream.Capacity = maxBufferLength; // make room for in-place Base64 conversion
+
+            _jwt = _memoryStream.GetBuffer();
+            _writer = null; // this will prevent subsequent additions of claims.
+
+            Span<byte> toEncode = _jwt.AsSpan(payloadIndex);
+            OperationStatus status = NS2Bridge.Base64UrlEncodeInPlace(toEncode, payloadLength, out int payloadWritten);
+            Debug.Assert(status == OperationStatus.Done); // Buffer is adjusted above, and so encoding should always fit
+
+            // Add signature
+            int headerAndPayloadLength = payloadWritten + headerSha256.Length;
+            _jwt[headerAndPayloadLength] = (byte)'.';
+            int headerAndPayloadAndSeparatorLength = headerAndPayloadLength + 1;
+
+            byte[] hashed;
+            HMAC hash = null;
+            switch (_algorithm)
+            {
+                case AccessTokenAlgorithm.HS256:
+                    hash = new HMACSHA256(_key);
+                    break;
+                case AccessTokenAlgorithm.HS512:
+                    hash = new HMACSHA512(_key);
+                    break;
+                default:
+                    break;
+            }
+            hashed = hash.ComputeHash(_jwt, 0, headerAndPayloadLength);
+            status = NS2Bridge.Base64UrlEncode(hashed, _jwt.AsSpan(headerAndPayloadAndSeparatorLength), out int consumend, out int signatureLength);
+            Debug.Assert(status == OperationStatus.Done); // Buffer is adjusted above, and so encoding should always fit
+            _jwtLength = headerAndPayloadAndSeparatorLength + signatureLength;
+
+            return _jwtLength;
+        }
+
+        public bool TryBuildTo(Span<char> destination, out int charsWritten)
+        {
+            End();
+            if (destination.Length < _jwtLength)
+            {
+                charsWritten = 0;
+                return false;
+            }
+            NS2Bridge.Latin1ToUtf16(_jwt.AsSpan(0, _jwtLength), destination);
+            charsWritten = _jwtLength;
+            return true;
+        }
+
+        public string BuildString()
+        {
+            End();
+            var result = NS2Bridge.CreateString(_jwtLength, _jwt, (destination, state) => {
+                NS2Bridge.Latin1ToUtf16(state.AsSpan(0, _jwtLength), destination);
+            });
+            return result;
+        }
+
+        // Modified from https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/b1814f6013262793512be596016d5c75bc5d4fea/src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs#L513
+        // ATTENTION: This method ONLY supports `Claim.Value` which is string. The original implementation can handle different `ClaimValueType`
```

Comment:
I think we need to support multiple ClaimValueTypes since it is a common contract